ci/jobs/pulp-upgrade.yaml

```diff
@@ -0,0 +1,164 @@
+                --skip erratum
+            pulp-admin rpm repo sync run --repo-id centos5 >/dev/null
+
+            pulp-admin rpm repo create \
+                --feed http://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/zoo \
+                --relative-url zoo \
+                --repo-id zoo \
+                --retain-old-count 0
+
+            pulp-admin rpm repo sync run --repo-id zoo
+            pulp-admin rpm repo sync schedules create \
+                --repo-id zoo \
+                --schedule 2015-06-06T14:29:00Z/PT1M
+
+            pulp-admin puppet repo create \
+                --feed http://forge.puppetlabs.com \
+                --repo-id forge
+            pulp-admin puppet repo sync run --repo-id forge >/dev/null
+
+            pulp-admin rpm repo create \
+                --feed http://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/pulp_unittest/ \
+                --repo-id repo_1
+
+            pulp-admin rpm repo create \
+                --feed file:///var/lib/pulp/published/yum/https/repos/repos/pulp/pulp/demo_repos/zoo/ \
+                --repo-id file-feed
+            pulp-admin rpm repo sync run --repo-id file-feed
+
+            pulp-admin iso repo create \
+                --feed https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/test_file_repo/ \
+                --repo-id filerepo
+            pulp-admin iso repo sync run --repo-id filerepo
+
+            if [ "$(echo -e \"2.8\n${{PULP_VERSION}}\" | sort -V | head -n 1)" = "2.8" ]; then
+                pulp-admin docker repo create \
+                    --feed https://index.docker.io \
+                    --repo-id busybox \
+                    --upstream-name busybox
+                pulp-admin docker repo sync run --repo-id busybox
+            fi
+
+            pulp-admin repo group create --group-id my-zoo --display-name "My Zoo"
+            pulp-admin repo group members add  --group-id my-zoo --repo-id zoo
+
+            pulp-admin rpm repo create \
+                --feed https://cdn.redhat.com/content/dist/rhel/rhui/server/7/7.2/x86_64/os/ \
+                --feed-ca-cert cdn/cdn.redhat.com-chain.crt \
+                --feed-cert cdn/914f702153514b06c1ef279db9dcadce.crt \
+                --feed-key cdn/914f702153514b06c1ef279db9dcadce.key \
+                --repo-id rhel7 \
+                --skip erratum
+            pulp-admin rpm repo sync run --repo-id rhel7 >/dev/null
+
+            pulp-admin logout
+        - shell: |
+            cat /etc/yum.repos.d/pulp.repo
+            sudo sed -i "s/${{PULP_VERSION}}/${{UPGRADE_PULP_VERSION}}/" /etc/yum.repos.d/pulp.repo
+            cat /etc/yum.repos.d/pulp.repo
+
+            sudo yum repolist
+
+            sudo yum -y update
+            sudo -u apache pulp-manage-db
+            sudo systemctl daemon-reload
+
+            for service in httpd pulp_workers pulp_celerybeat pulp_resource_manager; do
+                sudo systemctl restart "${{service}}"
+            done
```

Comment:
This loop can be simplified to:

    systemctl restart httpd pulp_workers pulp_celerybeat pulp_resource_manager

The more concise syntax is actually preferable, because systemctl can restart the units in parallel, and can take care of any ordering dependencies if any are present.